torch/_inductor/codegen/cpp.py

```diff
@@ -2589,24 +2589,28 @@ def reduction_combine_vec(
     def indirect_assert(self, var, lower, upper, mask=None):
         assert isinstance(var, CppCSEVariable)
         assert var.dtype is not None
-        if not var.is_vec:
+        if not (var.is_vec or (mask and mask.is_vec)):
             return super().indirect_assert(var, lower, upper, mask)
+        if var.is_vec:
+            var_str = str(var)
+        else:
+            var_str = f"{self._get_vec_type(var.dtype)}({var})"
```

Comment:
Aren't we missing the same logic for `mask`?
Also, we might want to add that logic to `cond`, not here, to perform the `cond` computation as a scalar.